### ISSUE TITLE
Integrate the React renderer into the generator

### ISSUE THREAD

**Issue by jonaslagoni** [asyncapi/shape-up-process#35]
It has to be backward compatible meaning that template configs should have an added `template` type where default (if not is defined) to Nunjucks.

**Comment by jonaslagoni** [comment]
/progress 5 Started figuring out how to integrate the react SDK into the generator.

**Comment by jonaslagoni** [comment]
/progress 55 Finished getting an overview on how to integrate react into the generator and refactored generator code to support multiple render engines while being backwards compatible. Waiting for the react SDK to finish before integrating the rest and creating tests.

**Comment by jonaslagoni** [comment]
/progress 65 Managed to successfully integrate the current react SDK and create a small test template. I also managed to get debugging (with vscode) to work where you could place breakpoints inside the react code and get triggered when you ran the generator. Still need to simplify the approach if possible to make it as smooth as possible for template owners.

**Comment by jonaslagoni** [comment]
/progress 70 Tried to solve some dependency issues using [rollup](https://github.com/rollup/rollup) which actually seems to work perfect for our scenario (i.e. bundle dependencies for each template file so we don't dictate where reusable components should be placed). Basically it is just a fancy wrapper for babel which minimizes some implementations from our side. This also solved some issues I had with debugging not triggering breakpoints (dont know why tbh, since its babel underneath, might be some timing issues. Gonna keep an eye out for it).

**Comment by jonaslagoni** [comment]
/progress 75 Multiple issues solved:
- Separated `nunjucks` and `react` code in the generator (that was a challenge, separate issue regarding refactoring needed, it is way too coupled). 
- Ensured that the template files in sub-directories within the template are transpiled only once instead of under each rendering and they get placed in correct folders (apparently `rollup` transpiles everything in the same directory which are not optimal for us). 
- Split out transpiler code from the rendering process to make it more maintaining.

**Next**: Adapt the `authoring.md` docs to include `react`. Create tests for the transpiler code. Docs for the SDK (which are generated with `jsdoc`) looks weird and needs to be fixed somehow.

**Comment by jonaslagoni** [comment]
/progress 90 Adapted `authoring.md`, added verification tests to generator, added transpiler tests, fixed docs for both generator and react SDK. Looks like it is about ready for a final review

**Comment by magicmatatjahu** [comment]
/progress 100 Done by https://github.com/asyncapi/generator/commit/fb6b41614a97f74b5cf3e507855513a8ef630031 commit
